source/validate_datarules.cpp

```diff
@@ -31,42 +31,43 @@ using libspirv::ValidationState_t;
 
 namespace {
 
-// vector types can only have 2 or 3 or 4 components.
-// if Vector16 capability is added, 8 and 16 components are also allowed.
+// Validates that the number of components in the vector type is legal.
```

Comment:
"legal" here is customary.  But personally I always write "valid" or "invalid" instead of "legal" vs. "illegal".  After all, the police will not come after you if you present an invalid module.  :-)

This is a personal quirk of mine.  
No need to change this CL.